pep8.py

```diff
@@ -1510,7 +1510,9 @@ def check_logical(self):
     def check_ast(self):
         """Build the file's AST and run all AST checks."""
         try:
-            tree = compile(''.join(self.lines), '', 'exec', PyCF_ONLY_AST)
+            source = ''.join(self.lines)
+            tree = compile(source, '', 'exec', PyCF_ONLY_AST)
+            tree.source = source
```

Comment:
Just running these lines at the command line (in Python 2.7.10) tree comes back as type `_ast.Module` which looks like it doesn't have a `source` attribute on it... Is this solely something that you make use of downstream to get back to the source in some fashion by monkeypatching the `source` onto the `tree`?

I think in either case I'd feel much more comfortable if an update to the docstring was added to clarify a bit.